pkg/guardian/redis_conf_store.go

```diff
@@ -42,8 +47,17 @@ func NewRedisConfStore(redis *redis.Client, defaultWhitelist []net.IPNet, defaul
 		limit:           defaultLimit,
 		reportOnly:      defaultReportOnly,
 		routeRateLimits: make(map[url.URL]Limit),
-	}
-	return &RedisConfStore{redis: redis, logger: logger, conf: &lockingConf{conf: defaultConf}, reporter: mr,}
+		jails:           make(map[url.URL]Jail),
+		// prisoners is a map of ips represented as string, to time.Time that represent the time, in utc, in which the entry in the map should expire.
+		// this is an artifact of 2 redis limitations:
+		// 1. Redis Hashmaps and Sets to not support expiry of individual keys
+		// 2. We could just rely upon unique keys in the global keyspace and do prefix lookups with the KEYS function, however, these lookups would be O(N).
+		//    N in this case is every single entry within the database. This is not a valid option considering we could have millions of keys and do not want to
+		//    block redis for too long.
+		prisoners: make(map[string]time.Time),
```

Comment:
It may be worth also storing the jail the prisoner is associated with.

Reply:
Agreed. It's times like these where I wish Guardian had access to a relational DB so that we can easily make these kinds of associations. 

Perhaps we could do a map[string]newStruct and just serialize/deserialize this struct when interacting with redis. Or go with the approach we have been, where we basically create a new redis hashmap to store this additional field within.